File: set2.c
// SET-2
#include <stdio.h>
#include <string.h>

char input[100];
int i=0;

void S();
void L();

void S(){
    if(input[i]=='a') i++;
    else if(input[i]=='('){
        i++; L();
        if(input[i]==')') i++;
        else printf("Error\n");
    } else printf("Error\n");
}

void L(){
    S();
    while(input[i]==','){
        i++; S();
    }
}

int main(){
    scanf("%s",input);
    S();

    if(input[i]=='\0') printf("Accepted\n");
    else printf("Rejected\n");

    char str[100]; scanf("%s",str);
    int len=strlen(str);
    if(str[len-1]=='0') printf("Even\n");
    else printf("Odd\n");
}

File: set2.py
# SET-2

s = input("Enter string: ")

# 1. Only allowed characters
for ch in s:
    if ch not in ['a', ',', '(', ')']:
        print("Rejected")
        exit()

# 2. Balanced parentheses
count = 0
for ch in s:
    if ch == '(':
        count += 1
    elif ch == ')':
        count -= 1
    if count < 0:
        print("Rejected")
        exit()

if count != 0:
    print("Rejected")
    exit()

# 3. Basic invalid patterns
if ",)" in s or "(," in s or s.endswith(','):
    print("Rejected")
else:
    print("Accepted")

# Odd binary
s = input().strip()
print("Odd" if s[-1]=='1' else "Even")
